src/lib/Microsoft.Health.Extensions.Fhir.R4/BundleExtensions.cs

```diff
@@ -96,30 +103,16 @@ public static async Task<IEnumerable<TResource>> ReadFromBundleWithContinuationA
                         count--;
                     }
                 }
+            };
+
+            storeResources.Invoke(bundle);
 
-                bundle = await fhirClient.ContinueAsync(bundle).ConfigureAwait(false);
+            await foreach (var currentBundle in fhirClient.IterateOverAdditionalBundlesAsync(bundle))

```

Comment:
I think that we should be passing in non-default Cancellation tokens into the fhir client. I've seen cases where calls to the Fhir Server using the original client hung. By enforcing the use of non-default cancellation tokens we can help mitigate that.

Reply:
Thanks for calling this out. We discussed this offline, and I've created a [WI](https://microsofthealth.visualstudio.com/Health/_workitems/edit/88541) that can be taken up in a subsequent PR to refactor upstream methods to pass non-default tokens. Currently, linking cancellation tokens in the FhirServiceRepository will prevent the FhirService from hanging